ui/src/shared/components/OptIn.js

```diff
@@ -0,0 +1,182 @@
+import React, {Component, PropTypes} from 'react'
+import classnames from 'classnames'
+
+import uuid from 'node-uuid'
+
+import ClickOutsideInput from 'shared/components/ClickOutsideInput'
+
+class OptIn extends Component {
+  constructor(props) {
+    super(props)
+
+    const {customValue, fixedValue} = props
+
+    this.state = {
+      useCustomValue: customValue !== '',
+      fixedValue,
+      customValue,
+    }
+
+    this.id = uuid.v4()
+    this.isCustomValueInputFocused = false
+
+    this.useFixedValue = ::this.useFixedValue
+    this.useCustomValue = ::this.useCustomValue
+    this.considerResetCustomValue = ::this.considerResetCustomValue
+    this.setCustomValue = ::this.setCustomValue
+    this.setValue = ::this.setValue
+  }
+
+  useFixedValue() {
+    this.setState({useCustomValue: false, customValue: ''}, () =>
+      this.setValue()
+    )
+  }
+
+  useCustomValue() {
+    this.setState({useCustomValue: true}, () => this.setValue())
+  }
+
+  handleClickFixedValueField() {
+    return () => this.useFixedValue()
+  }
```

Comment:
Looks like this function is unnecessary.  Why not call `useFixedValue` directly in the `onClick` below?